sources/Renderer/Direct3D11/D3D11SwapChain.cpp

```diff
@@ -260,18 +267,49 @@ static UINT GetPrimaryDisplayRefreshRate()
 
 void D3D11SwapChain::CreateSwapChain(IDXGIFactory* factory, const Extent2D& resolution, std::uint32_t samples, std::uint32_t swapBuffers)
 {
-    /* Get current settings */
-    const DXGI_RATIONAL refreshRate{ GetPrimaryDisplayRefreshRate(), 1 };
-
     /* Pick and store color format */
     colorFormat_ = DXGI_FORMAT_R8G8B8A8_UNORM;//DXGI_FORMAT_B8G8R8A8_UNORM
+    
+    /* Create swap chain for window handle */
+    NativeHandle wndHandle = {};
+    GetSurface().GetNativeHandle(&wndHandle, sizeof(wndHandle));
+
+#if LLGL_D3D11_ENABLE_FEATURELEVEL >= 3
+    /* Clamp buffer count between 2 and max buffers */
+    swapBuffers = std::max(2u, std::min<std::uint32_t>(swapBuffers, DXGI_MAX_SWAP_CHAIN_BUFFERS));
+
+    /* Find suitable multi-samples for color format */
+    swapChainSampleDesc_ = D3D11RenderSystem::FindSuitableSampleDesc(device_.Get(), colorFormat_, 1); // TODO: resolve multi-sampling
+
+    DXGI_SWAP_CHAIN_DESC1 swapChainDesc = {};
+    {
+            swapChainDesc.Width         = resolution.width;
+            swapChainDesc.Height        = resolution.height;
+            swapChainDesc.Format        = colorFormat_;
+            swapChainDesc.SampleDesc    = swapChainSampleDesc_;
+            swapChainDesc.BufferUsage   = DXGI_USAGE_RENDER_TARGET_OUTPUT;
+            swapChainDesc.BufferCount   = swapBuffers;
+            swapChainDesc.SwapEffect    = DXGI_SWAP_EFFECT_FLIP_DISCARD; // FLIP effect requires BufferCount >= 2 && SampleDesc.Count == 1
+            swapChainDesc.Flags         = (tearingSupported_ ? DXGI_SWAP_CHAIN_FLAG_ALLOW_TEARING : 0u);
+    }
+
+    ComPtr<IDXGIFactory2> factory2;
+    HRESULT hr = factory->QueryInterface(IID_PPV_ARGS(&factory2));
```

Comment:
Just for the factory query, I think it makes sense to move this upward within this `#if` block and if it fails, fallback to the alternative swap-chain creation, just like you did in `D3D11SwapChain::CheckTearingSupport`. If `CreateSwapChainForHwnd` fails with a valid factory, we should indeed crash with a callstack.

Reply:
Makes sense, will do that